lesson_20/homework_20.py

```diff
@@ -0,0 +1,119 @@
+"""Homework 20."""
+import csv
+import logging
+import sqlite3
+
+"""
+Створіть базу даних для інтернет-магазину з наступними таблицями:
+products: таблиця для зберігання інформації про продукти,
+включаючи назву, опис, ціну тощо.
+categories: таблиця для категорій продуктів.
+products повинна мати зовнішній ключ на таблицю categories.
+Напишіть SQL-скрипт для створення зазначених таблиць.
+Внесіть декілька рядків даних в кожну таблицю
+Виконайте JOIN-запит, який повертає інформацію про продукти
+та назву їх категорій
+"""
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def create_tables():
+    """Create tables."""
+    with sqlite3.connect('internet_shop.db') as conn:
+        cursor = conn.cursor()
+        cursor.execute("""
+            CREATE TABLE IF NOT EXISTS categories (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                name TEXT NOT NULL
+            )
+        """)
+        cursor.execute("""
+            CREATE TABLE IF NOT EXISTS products (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                name TEXT NOT NULL,
+                description TEXT,
+                price INTEGER NOT NULL CHECK (price >= 0),
+                category_id INTEGER,
+                FOREIGN KEY (category_id) REFERENCES categories (id)
+            )
+        """)
+        logging.info('Tables created successfully.')
+
+
+def populate_tables_from_csv(categories_csv, products_csv):
+    """Populate the table from CSV files."""
+    with sqlite3.connect('internet_shop.db') as conn:
+        cursor = conn.cursor()
+        try:
+            # Populate categories
+            with open(categories_csv, 'r') as cat_file:
+                reader = csv.DictReader(cat_file)
+                categories = [(row['name'],) for row in reader]
```

Comment:
Reading data from csv files could be organized as separate function with independent exception handling